L1Trigger/L1THGCal/plugins/HGCalTriggerDigiProducer.cc

```diff
@@ -39,9 +39,9 @@ DEFINE_FWK_MODULE(HGCalTriggerDigiProducer);
 
 HGCalTriggerDigiProducer::
 HGCalTriggerDigiProducer(const edm::ParameterSet& conf):
-  inputee_(consumes<HGCEEDigiCollection>(conf.getParameter<edm::InputTag>("eeDigis"))),
-  inputfh_(consumes<HGCHEDigiCollection>(conf.getParameter<edm::InputTag>("fhDigis"))), 
-  inputbh_(consumes<HGCBHDigiCollection>(conf.getParameter<edm::InputTag>("bhDigis"))), 
+  inputee_(consumes<HGCalDigiCollection>(conf.getParameter<edm::InputTag>("eeDigis"))),
```

Comment:
Can it still read old samples containing `HGCEEDigiCollection`, etc. and not `HGCalDigiCollection`? Right now I get:
```
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::SortedCollection<HGCDataFrame<DetId,HGCSample>,edm::StrictWeakOrdering<HGCDataFrame<DetId,HGCSample> > >
Looking for module label: mix
Looking for productInstanceName: HGCDigisEE
```

Reply:
No, it will no longer read old samples. (The type is technically different, even though the data stored are the same.) Old samples can still be read in old releases. If this is a problem, let me know.

Reply:
Normally the new code should remain able to read data produced with the previous versions of it, unless there are no user case for it. Apparently those user cases do exist, and therefore the back-compatibility should better be granted here

Reply:
My goal here was to avoid littering the code with old version / new version flags (just because of a poor design choice from years ago)... let me think about how to handle this.

Reply:
What about a converter module that can be added in the L1T reprocessing sequences? It would just take in digis of the old type and output digis of the new, more generic type (the underlying data are the same).

Reply:
Would that module run in all versions, and it does nothing when the new digis data format is given in input? That may do the job, and it can be removed later when there will be no need any more to read old data samples

Reply:
@perrotta I'm not sure if we could insert it into the default sequences sensibly. The digis are consumed with this kind of input tag: `eeDigis = cms.InputTag('mix:HGCDigisEE')`, and there is already a `process.mix` producer in the sequences.

In any case, in central production we run DIGI and L1 steps together, so from now on, they will always get the new collection type. It is only for a user job trying to read old files that the conversion would be needed. My plan was to provide a customize function so user configs could be updated easily and (relatively) painlessly. The customize function would add the converter and modify input tags appropriately.

Reply:
An intermediate converter module + customization function should do the job, and it will be easy from the user side to switch between old and new samples.
Thanks!